src/server/game/Movement/Spline/MovementPacketBuilder.cpp

```diff
@@ -98,7 +98,7 @@ namespace Movement
 
     void WriteLinearPath(const Spline<int32>& spline, ByteBuffer& data)
     {
-        uint32 last_idx = spline.getPointCount() - 3;
+        uint32 last_idx = spline.getPointCount() - 2;
```

Comment:
You just pass all points you see in the packet (and final destination)
First and last points in spline are internal generated points (yes, catmullrom init is used for linear splines too!), that makes it -2 points to send. On top of these points one is sent separately (destination), making it -3